### ISSUE TITLE
Add diffusion model RM

### ISSUE THREAD

**Issue by shahbuland** [CarperAI/magicarp-v2#13]
Need a diffusion model RM that takes variance schedule as input. Taking pretrained cross-attention DM and finetuning with a reward head at the UNET bottleneck (before upsampling blocks) should suffice.

**Comment by shahbuland** [comment]
Started work in diffuser-rm branch. What exactly is needed:
- CrossEncoder that wraps around `magicarp.models.nn.hidden_state_unet.Unet2DConditionalHidden`
- Pipeline that makes use of VAE from a diffusers checkpoint and applies noise according to the noise schedule
-> idea is that subsequent pipelines for concrete tasks can inherit from above pipeline
